sdk/scheduler/src/main/java/com/mesosphere/sdk/config/ConfigurationUpdater.java

```diff
@@ -20,16 +20,35 @@
          * The resulting configuration ID which should be used by service tasks.
          */
         public final UUID targetId;
+        private final UpdateType updateType;
+
+        /**
+         * Two types of deployments are differentiated by this type.  Either a services is
+         * being deployed for the first time "DEPLOY", or it is updating a previously deployed
+         * version of the service, "UPDATE"
+         */
+        public enum UpdateType {
+            DEPLOY,
+            UPDATE
+        }
 
         /**
          * A list of zero or more validation errors with the current configuration. If there were
          * errors, the {@link #targetId} will point to a previous valid configuration.
          */
         public final Collection<ConfigValidationError> errors;
 
-        public UpdateResult(UUID targetId, Collection<ConfigValidationError> errors) {
+        public UpdateResult(
+                UUID targetId,
+                UpdateType updateType,
+                Collection<ConfigValidationError> errors) {
             this.targetId = targetId;
             this.errors = errors;
+            this.updateType = updateType;
+        }
+
+        public UpdateType getUpdateType() {
+            return  updateType;
```

Comment:
double space

while in here, consider switching to accessors for targetId/errors as well